src/useA11yBlockLink/useA11yBlockLink.tsx

```diff
@@ -0,0 +1,50 @@
+interface EventObject {
+  target: {
+    localName: string;
+    href: string;
+    type: string;
+    role: string;
+  };
+}
+interface RefProps {
+  href?: string | null;
+  click?: () => {};
+  current?: RefProps;
+}
+
+export default function useA11yClick(props: {mainClick: RefProps}) {
+  function handleClick(e: EventObject) {
+    // Confirm that a main click property is provided to handle the click accessibility.
+    if (!props.mainClick) {
+      throw Error(
+        '`useA11yClick` requires a `mainClick` handle the click around the entire container.'
+      );
+    }
```

Comment:
Nitpick: (feel free to drop if you want to)

I'd recommend moving this block up earlier to throw in the main body of the hook itself, that way it throws eagerly instead of lazily (e.g. right now it will throw when the user clicks instead of when the developer calls the hook)